src/compiler/checker.ts

```diff
@@ -34625,7 +34623,6 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             case SyntaxKind.JsxSelfClosingElement:
                 return resolveJsxOpeningLikeElement(node, candidatesOutArray, checkMode);
         }
-        Debug.assertNever(node, "Branch in 'resolveSignature' should be unreachable.");

```

Comment:
I'm not sure why you're removing code; this change should probably only include changes to diagnostic messages (and their references) and that's it.

Reply:
Ooh, I really don't know how this slipped through the cracks. I actually just wanted to adjust the imports. Could automatic code formatting have done this? I will quickly revert.

Reply:
I'm not aware of any tool that would have done that 😅

Reply:
It shows "unreachable code detected" in my VS Code, so I guess it was removed when I applied quick fixes. Sorry for this.

![image](https://github.com/microsoft/TypeScript/assets/469989/c9d6af6f-cf47-403a-9b72-29c3984b97c6)

Reply:
I could verify my suspicion. I had enabled the "format on save" feature, and simply by saving the file, the `break` statement was removed due to it being unreachable code. I initially missed this due to my Git UI indicating that the file exceeds the limit of 1000000 bytes. I'll make adjustments to my configuration. I apologize for the oversight once again.

![image](https://github.com/microsoft/TypeScript/assets/469989/8a48b9cc-0a82-44e6-b969-3d5955221ef3)

Reply:
I've never heard of format on save doing this; do you have some other extension that could be doing this?

Reply:
If this were common, you could accidentally write `return` at the top of the function and it'd delete the entire thing! So, that sounds like a problem.

Reply:
@jakebailey I actually have seen this happen. My first suspicion was that it was our `eslint.codeActionsOnSave.rules`, but the config looks like this:

```json
	"editor.defaultFormatter": "esbenp.prettier-vscode",
	"editor.formatOnSave": true,
	"editor.codeActionsOnSave": ["source.fixAll.eslint"],
	"eslint.codeActionsOnSave.rules": [
		"import/order",
		"sort-imports",
		"import/no-duplicates",
		"@typescript-eslint/consistent-type-imports",
		"object-shorthand",
		"prefer-arrow-functions/prefer-arrow-functions"
	],
```

So now I'm not sure what would have caused the unreachable code to be deleted in the cases I've seen it happen.